release-notes.txt

```diff
@@ -1,3 +1,767 @@
+RIOT-2022.07 - Release Notes
+============================
+RIOT is a multi-threading operating system which enables soft real-time
+capabilities and comes with support for a range of devices that are typically
+found in the Internet of Things: 8-bit and 16-bit microcontrollers as well as
+light-weight 32-bit processors.
+
+RIOT is based on the following design principles: energy-efficiency, soft
+real-time capabilities, small memory footprint, modularity, and uniform API
+access, independent of the underlying hardware (with partial POSIX compliance).
+
+RIOT is developed by an international open-source community which is
+independent of specific vendors (e.g. similarly to the Linux community) and is
+licensed with a non-viral copyleft license (LGPLv2.1), which allows indirect
+business models around the free open-source software platform provided by RIOT.
+
+
+About this release
+==================
+
+The 2022.07 release includes many changes, some highlights of which are:
+
+### GPIO Low-level drivers
+
+A new API, `gpio_ll`, has been introduced for highly efficient access to GPIO
+pins grouped in ports.
+
+The API is limited to internal use for the time being, but still paves the way
+towards using port multipliers like local GPIO pins, and for improved
+bit-banging implementations (e.g. around WS281x-style LEDs).
+
+### Improved serial port selection
+
+The automatic selection of serial ports has been improved by considering more
+metadata. When using multiple board of different kinds simultaneously, passing
+the `PORT` variable will often not be required any more. When disambiguation is
+still not possible, setting `MOST_RECENT_PORT=1` often helps picking the right
+one.
```

Comment:
Note that this is not entirely true. The feature is only made use of when `MOST_RECENT_PORT=1` is set. With `MOST_RECENT_PORT=1` the list of serials if first filtered to match only the serials that may belong to the used board (if the board provides such filters), and afterwards the most recently connected one of them is picked.

The issues is that the python script depends on a python module that may not be installed on any system and I rather did not like to force in an additional dependency for a pure convenience feature.

Reply:
Hm, the weird thing is that as I started rewriting that paragraph, it sounded like if your serial setup was complicated for any reason, you'd just start using `MOST_RECENT_PORT=1` -- even if it's not the most recent one but just the most recent one you want, you just want to use the new discovery methods.

Maybe that could get some improvement and a ready-for-prime-time announcement in the next release?

Reply:
It would certainly be more useful if more boards would add proper filters. But only for boards with integrated serials this makes sense.

@bergzand With serial over CDC-ACM we should be able to provide the RIOT board name, shouldn't we? Right now I see


path         | driver  | vendor              | model        | model_db                                                      | serial           | ctime    | iface_num
-------------|---------|---------------------|--------------|---------------------------------------------------------------|------------------|----------|----------
/dev/ttyACM1 | cdc_acm | RIOT-os.org         | USB device   | 100 Series/C230 Series Chipset Family USB 3.0 xHCI Controller | 6BB4583D2FB09E81 | 15:37:59 | 0        

(Isn't it convenient that the output of `make list-ttys` is valid markdown :smile:)

It would be super convenient if the model would be e.g. `weact-401cc` for my `weact-401cc` board instead. That would be super easy to filter against :)

Reply:
(And yes, using a virtual env here may be a solution to just enable this by default.)

Reply:
Updated in ef4351e9b70ef3cae72cfcfea8beabd00e230011 -- is this now correct?